src/libNode/PoWProcessing.cpp

```diff
@@ -107,36 +107,7 @@ bool Node::StartPoW(const uint64_t& block_num, uint8_t ds_difficulty,
   LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
             "Current dsblock is " << block_num);
 
-  bool isNoSync = false;
-
-  if (m_mediator.m_lookup->m_syncType == SyncType::NO_SYNC) {
-    m_stillMiningPrimary = true;
-    isNoSync = true;
-
-    auto func = [this]() mutable -> void {
-      this_thread::sleep_for(chrono::seconds(NEW_NODE_SYNC_INTERVAL +
-                                             POW_WINDOW_IN_SECONDS +
-                                             FALLBACK_EXTRA_TIME));
-      if (m_stillMiningPrimary) {
-        if (!GetOfflineLookups()) {
-          LOG_GENERAL(WARNING, "Cannot sync currently");
-          return;
-        }
-
-        if (GetLatestDSBlock()) {
-          LOG_GENERAL(
-              INFO,
-              "New DS Block mined but I'm still mining and didn't receive it,"
-              " rejoin");
-          RejoinAsNormal();
-        } else {
-          LOG_GENERAL(INFO, "Didn't get the latest DSBlock, what to do???");
-        }
-      }
-    };
-
-    DetachedFunction(1, func);
-  }
```

Comment:
Why to remove this part? Is it causing problem?

Reply:
Yes, it will also cause rejoin, but now if wait for DS block timeout then trigger rejoin already.

Reply:
This check here is for different purpose, if node is still mining for a long duration. The other thread only activates when node has finished mining.

Reply:
Then we should check the mining time, if the mining takes too long, then should stop mining. In my local test case, this piece of code make node always fetching latest Tx blocks, then it need another fix.